src/gravel/controllers/nodes.py

```diff
@@ -163,7 +171,17 @@ def __init__(self):
         self._incoming_task = asyncio.create_task(self._incoming_msg_task())
 
     async def join(self, address: str, token: str) -> bool:
-        logger.debug(f"=> mgr -- join > with addr {address}, token: {token}")
+        logger.debug(f"=> mgr -- join > with leader {address}, token: {token}")
+
+        if self._state:
+            assert self._state.role == NodeRoleEnum.LEADER or \
+                   self._state.role == NodeRoleEnum.FOLLOWER
```

Comment:
(nit) wouldn't it be easier and more future proof to check that `self._state.role == NodeRoleEnum.none`?